packages/office-ui-fabric-react/src/components/VirtualizedList/VirtualizedList.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from 'react';
+import { BaseComponent, css, findScrollableParent, getParent, autobind } from '../../Utilities';
+import { IObjectWithKey } from '../../utilities/selection/index';
+import { IVirtualizedListProps } from './VirtualizedList.Props';
+import { IScrollContainerContext, ScrollContainerContextTypes } from '../../utilities/scrolling/ScrollContainer';
+
+interface IRange {
+  /** Start of range */
+  start: number;
+
+  /** Exclusive end of range */
+  end: number;
+}
+
+function isInRange(range: IRange, index: number): boolean {
+  return range.start <= index && index < range.end;
+}
+
+export interface IVirtualizedListState {
+  viewportHeight: number;
+
+  items: React.ReactNode[];
+}
+
+export class VirtualizedList<TItem extends IObjectWithKey> extends BaseComponent<IVirtualizedListProps<TItem>, IVirtualizedListState> {
+  public static contextTypes = ScrollContainerContextTypes;
+  public context: IScrollContainerContext;
+
+  private _root: HTMLElement;
+
+  private _spacerElements: { [key: string]: HTMLElement } = {};
+
+  private _focusedIndex: number = -1;
+
+  constructor(props: IVirtualizedListProps<TItem>, context: IScrollContainerContext) {
+    super(props, context);
+
+    const {
```

Comment:
I'm trying to avoid measurements (forced layouts) for the initial rendering. Thought drawing more than required here is better than messing with the initial page layout